src/groupByKey.ts

```diff
@@ -2,6 +2,21 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T extends object>(
+  items: T[],
+  key: keyof T,
+): GroupsMap<T> {
+  return items.reduce((acc, item) => {
+    const value = item[key];
+
+    if (typeof value === 'string' || typeof value === 'number') {
+      if (!acc[value]) {
+        acc[value] = [];
+      }
+
+      acc[value].push(item);
+    }
```

Comment:
The check for 'string' or 'number' is good for ensuring the key values are valid for object properties. However, it's worth noting that if the 'key' argument passed to the function can only be a property of T, and T extends object, the types of the properties should already be enforced by TypeScript's type system. If you're expecting keys with values other than strings or numbers, consider handling them or explicitly stating the limitation in the function documentation.